plugin-server/src/kafka/consumer.ts

```diff
@@ -369,11 +406,37 @@ export class KafkaConsumer {
                         histogramKafkaConsumeInterval.labels({ topic, groupId }).observe(intervalMs)
                     }
                     lastConsumeTime = consumeStartTime
-                    // TRICKY: We wrap this in a retry check. It seems that despite being connected and ready, the client can still have an undeterministic
-                    // error when consuming, hence the retryIfRetriable.
-                    const messages = await retryIfRetriable(() =>
-                        promisifyCallback<Message[]>((cb) => this.rdKafkaConsumer.consume(this.fetchBatchSize, cb))
-                    )
+
+                    let messages: Message[]
+
+                    if (this.config.continuousConsumptionEnabled) {
```

Comment:
question: instead of duplicating the fetch code, could we reuse the `continuousConsume` function somehow? We could call it before pulling messages from the queue and pass some sort of a `fromMainLoop` parameter to not do the reading in main loop check.